### PR TITLE
Parse `export` statements; fix comma parsing

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

- Parse `export` statements similarly to `import`
- Comma operator is now treated specially wrt. spacing; the typical `Foo, Bar` usage no longer creates operator sections.

### Important Notes

- Commas creating operator sections was super weird (it's just what happens if you treat comma like other operators). @JaroslavTulach Any time you notice something in my output that doesn't make any sense, please tell me! The specification I'm working from only specifies what inputs the parser should accept; I'm trying to keep the representations semantically reasonable but I will miss some cases like this.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

[ci no changelog needed]

### CODE REVIEW

File: lib/rust/parser/generate-java/java/org/enso/syntax2/Message.java [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
+
 final class Message {
     private final java.nio.ByteBuffer buffer;
     private final java.nio.ByteBuffer context;
     private final int base;
+    private final long metadata;
     private boolean encounteredUnsupportedSyntax;
 
-    public Message(java.nio.ByteBuffer bufferIn, java.nio.ByteBuffer contextIn, long baseIn) {
+    public Message(java.nio.ByteBuffer bufferIn, java.nio.ByteBuffer contextIn, long baseIn, long metadataIn) {
```
Comment (JaroslavTulach) [inline lib/rust/parser/generate-java/java/org/enso/syntax2/Message.java:13]: Methods in this non-public class don't have to be `public`.

**Review by JaroslavTulach — APPROVED** [review]

**Comment by JaroslavTulach** [comment]
> Any time you notice something in my output that doesn't make any sense, please tell me!

It is not always easy to recognize when the `Tree` is constructed just by an accident, but I'll do my best.

**Comment by kazcw** [comment]
This includes #3653.

**Comment by kazcw** [comment]
> > Any time you notice something in my output that doesn't make any sense, please tell me!
> 
> It is not always easy to recognize when the `Tree` is constructed just by an accident, but I'll do my best.

I just don't want you to have to do any extra work to deal with silly parses :smile:. There must be some code in AstToIr that takes apart a tree of comma operator-sections and builds a simple sequence from them--that should not exist! If it was inherited from dealing with the same bug in the old parser, I'm glad to make it legacy.

**Comment by kazcw** [comment]
Rolling this into #3681.

File: lib/rust/parser/tests/parse.rs [hunk 2]
```diff
@@ -471,6 +471,91 @@ fn import() {
     cases.into_iter().for_each(|(code, expected)| test(code, expected));
 }
 
+#[test]
+fn export() {
+    #[rustfmt::skip]
+    let cases = [
+        ("export Foo", block![(Export () () ((Ident export) (Ident Foo)) () ())]),
```
Comment (JaroslavTulach) [inline lib/rust/parser/tests/parse.rs:478]: `export Foo` isn't valid in current parser. The current parser needs FQN: `export prj.Data.Foo` is OK.

File: lib/rust/parser/tests/parse.rs [hunk 3]
```diff
@@ -471,6 +471,91 @@ fn import() {
...
+             ((Ident export) (OprApp (Ident Bar) (Ok ",") (Ident Baz)))
+             () ())]),
+        ("from Foo export all hiding Bar, Baz", block![
+            (Export
+             ((Ident from) (Ident Foo))
+             ()
+             ((Ident export) (Ident all))
+             ()
+             ((Ident hiding) (OprApp (Ident Bar) (Ok ",") (Ident Baz))))]),
+        ("from Foo as Bar export Baz, Quux", block![
```
Comment (JaroslavTulach) [inline lib/rust/parser/tests/parse.rs:494]: `from Foo as Bar export`!? Not sure what that can mean... why should I rename `Foo` to `Bar` when I export something else?

Comment (kazcw) [reply lib/rust/parser/tests/parse.rs:494]: I'm just supporting what is used in the libs. This pattern occurs in `distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex/Engine.enso`.

File: lib/rust/parser/tests/parse.rs (hunk 2 above)
Comment (kazcw) [reply lib/rust/parser/tests/parse.rs:478]: I see. In the next PR I'll change it to `prj.Data.Foo`.

**Pull request closed without merging**